### PR TITLE
[REF] 8.0 mgmtsystem claim

### PR DESCRIPTION
**Pull request by llacroix**
Fixed a few bugs about the stage_id yet someone should have a look to make sure everything is fine. I also updated some part of the models to API 8.0

Depends on:
- [x] #20

### CODE REVIEW

File: mgmtsystem_claim/board_mgmtsystem_claim.xml [hunk 1]
```diff
@@ -3,11 +3,13 @@
     <data>
 
         <record model="ir.actions.act_window" id="action_mgmtsystem_claim">
+	    <field name="name">board.mgmtsystem.claim.act_window</field>
             <field name="res_model">mgmtsystem.claim</field>
             <field name="view_type">form</field>
             <field name="view_mode">tree</field>
             <field name="view_id" ref="mgmtsystem_claims_tree_view"/>
-            <field name="domain">[('state', 'in', ('draft','open','pending'))]</field>
+	    <!-- TODO static list might not match stages -->
```
Comment (bwrsandman) [inline mgmtsystem_claim/board_mgmtsystem_claim.xml]: TODO, is this a work in progress?

File: mgmtsystem_claim/mgmtsystem_claim.py [hunk 2]
```diff
@@ -19,35 +19,21 @@
...
-        'company_id': fields.many2one('res.company', 'Company')
-    }
 
-    _defaults = {
-        'company_id': (
-            lambda self, cr, uid, c:
-            self.pool.get('res.users').browse(cr, uid, uid, c).company_id.id),
-        'reference': 'NEW',
-    }
+    reference = fields.Char('Reference', size=64, require=True, readonly=True,
```
Comment (bwrsandman) [inline mgmtsystem_claim/mgmtsystem_claim.py]: Is the size necessary?
typo in required

File: mgmtsystem_claim/mgmtsystem_claim.xml [hunk 3]
```diff
@@ -27,6 +30,7 @@
 				<button name="case_close" string="Close"
 					states="open,draft,pending" type="object"
 					icon="gtk-close" />
+		                -->
```
Comment (bwrsandman) [inline mgmtsystem_claim/mgmtsystem_claim.xml]: Better to remove than to comment

File: mgmtsystem_claim/mgmtsystem_claim.xml [hunk 4]
```diff
@@ -37,10 +41,11 @@
 		<field name="arch" type="xml">
 			<form string="Claim" version="7.0">
                 <header>
+			<!--
                     <button name="case_close" string="Settle" type="object" class="oe_highlight"
                             states="draft,open,pending" groups="base.group_user"/>
                     <button name="case_cancel" string="Reject" type="object" groups="base.group_user"
-                            states="draft,open,pending"/>
+			    states="draft,open,pending"/> -->
```
Comment (bwrsandman) [inline mgmtsystem_claim/mgmtsystem_claim.xml]: Better to remove than to comment

File: mgmtsystem_claim/mgmtsystem_claim.xml [hunk 5]
```diff
@@ -55,7 +60,7 @@
                         <field name="priority"/>
                         <field name="section_id"/>
                         <field name="date_deadline"/>
-                        <field name="state" groups="base.group_no_one"/>
+			<!-- <field name="state" groups="base.group_no_one"/> -->
```
Comment (bwrsandman) [inline mgmtsystem_claim/mgmtsystem_claim.xml]: Better to remove than to comment

File: mgmtsystem_claim/mgmtsystem_claim.xml [hunk 6]
```diff
@@ -143,7 +149,7 @@
 				<filter icon="terp-gtk-media-pause"
 					string="Pending"
 					domain="[('state','=','pending')]"
-					separator="1" help="All pending Claims" />
+					separator="1" help="All pending Claims" /> -->
```
Comment (bwrsandman) [inline mgmtsystem_claim/mgmtsystem_claim.xml]: Better to remove than to comment

File: mgmtsystem_claim/mgmtsystem_claim.xml [hunk 7]
```diff
@@ -165,9 +171,9 @@
 						domain="[]" context="{'group_by':'stage_id'}" />
 					<filter string="Type" icon="terp-stock_symbol-selection"
 						domain="[]" context="{'group_by':'categ_id'}" />
-					<filter string="State"
+					<!-- <filter string="State"
 						icon="terp-stock_effects-object-colorize"
-						domain="[]" context="{'group_by':'state'}" />
+						domain="[]" context="{'group_by':'state'}" />-->
```
Comment (bwrsandman) [inline mgmtsystem_claim/mgmtsystem_claim.xml]: Better to remove than to comment

**Comment by dreispt** [comment]
The conversion from States to Stages will mean a deeper change, and IMO it's best to to that after the port. In the meanwhile I'm OK with keeping the commented code to remind us of that.

**Comment by llacroix** [comment]
That's what I though, so keeping the commented things here? I'm aware that stages will have to be handled differently and we loose some functionality at the moment. If there is anything particularly wrong let me know and I'll fix it. 

I'll fix the errors with flake8

**Comment by dreispt** [comment]
As per [this comment](https://github.com/OCA/management-system/pull/22#issuecomment-64170867) my opinion is to keep them. But you still have a typo to fix, reported by @bwrsandman.

**Comment by bwrsandman** [comment]
[Coverage has gone down 10%](https://coveralls.io/files/365380716), please add a test of mgmtsystem_claim/mgmtsystem_claim.py `create` function

**Comment by llacroix** [comment]
Done

**Comment by bwrsandman** [comment]
@llacroix anything to be done with the TODOs and comments?
If so, please make an issue so we can have it in the 8.0 milestone.

**Comment by llacroix** [comment]
I added an issue here: https://github.com/OCA/management-system/issues/41

**Comment by llacroix** [comment]
Rebased

**Comment by dreispt** [comment]
I believe the TODO comments on code are helpful, and IMO the stage/state thing can be handled in a later PR.
So: :+1:

**Comment by llacroix** [comment]
@dreispt anything on this PR?

**Comment by bwrsandman** [comment]
@llacroix there is still some dead code

File: mgmtsystem_claim/mgmtsystem_claim.xml [hunk 8]
```diff
@@ -7,7 +7,9 @@
 		<field name="name">Management System Claims Tree</field>
 		<field name="model">mgmtsystem.claim</field>
 		<field name="arch" type="xml">
-			<tree string="Claims" colors="blue:state=='pending' and not(date_deadline and (date_deadline &lt; current_date));gray:state in ('close', 'cancel');red:date_deadline and (date_deadline &lt; current_date)">
+			<tree string="Claims" >
+				<!-- TODO use stage_id to change color -->
```
Comment (bwrsandman) [inline mgmtsystem_claim/mgmtsystem_claim.xml]: Is this coment to say that if you want colors, uncomment the line bellow?

If that is the case, it should be worded appropriately, `TODO` gives the impression that there is dev work to be done.

File: mgmtsystem_claim/mgmtsystem_claim.xml [hunk 9]
```diff
@@ -7,7 +7,9 @@
 		<field name="name">Management System Claims Tree</field>
 		<field name="model">mgmtsystem.claim</field>
 		<field name="arch" type="xml">
-			<tree string="Claims" colors="blue:state=='pending' and not(date_deadline and (date_deadline &lt; current_date));gray:state in ('close', 'cancel');red:date_deadline and (date_deadline &lt; current_date)">
+			<tree string="Claims" >
```
Comment (bwrsandman) [inline mgmtsystem_claim/mgmtsystem_claim.xml]: The red colour can be put here: `colors="red:date_deadline and (date_deadline &lt; current_date)"`

**Comment by dreispt** [comment]
Done my part here; we're waiting on a second approval.

**Comment by llacroix** [comment]
@dreispt here needs for new review
@bwrsandman here

**Comment by bwrsandman** [comment]
:+1: Good work!

**Pull request merged**